src/main/java/uk/gov/hmcts/dts/fact/services/admin/list/AdminAreasOfLawService.java

```diff
@@ -19,10 +23,63 @@ public AdminAreasOfLawService(final AreasOfLawRepository areasOfLawRepository) {
         this.areasOfLawRepository = areasOfLawRepository;
     }
 
+    public AreaOfLaw getAreaOfLaw(final Integer id) {
+        try {
+            return new AreaOfLaw(areasOfLawRepository.getOne(id));
+        } catch (final javax.persistence.EntityNotFoundException exception) {
+            throw new NotFoundException(exception);
+        }
+    }
+
     public List<AreaOfLaw> getAllAreasOfLaw() {
         return areasOfLawRepository.findAll()
             .stream()
             .map(AreaOfLaw::new)
             .collect(toList());
     }
+
+    @Transactional
+    public AreaOfLaw updateAreaOfLaw(final AreaOfLaw updatedAreaOfLaw) {
+        // Ensure entity with ID exists
+        final Optional<uk.gov.hmcts.dts.fact.entity.AreaOfLaw> areaOfLawEntity =
+            areasOfLawRepository.findById(updatedAreaOfLaw.getId());
+        if (areaOfLawEntity.isEmpty()) {
+            throw new NotFoundException(updatedAreaOfLaw.getId().toString());
+        }
+
+        // Update area of law entity
+        final uk.gov.hmcts.dts.fact.entity.AreaOfLaw entity = updateEntityPropertiesFromModel(updatedAreaOfLaw, areaOfLawEntity.get());
+        return new AreaOfLaw(areasOfLawRepository.save(entity));
+    }
+
+    @Transactional
+    public AreaOfLaw createAreaOfLaw(final AreaOfLaw areaOfLaw) {
+        checkIfAreaOfLawAlreadyExists(areaOfLaw.getName());
+        return new AreaOfLaw(areasOfLawRepository.save(createNewAreaOfLawEntityFromModel(areaOfLaw)));
+    }
+
+    private void checkIfAreaOfLawAlreadyExists(final String areaOfLawName) {
+        if (getAllAreasOfLaw().stream().anyMatch(aol -> aol.getName().equalsIgnoreCase(areaOfLawName))) {
+            throw new DuplicatedListItemException("Area of Law already exists: " + areaOfLawName);
+        }
+    }
+
+    private uk.gov.hmcts.dts.fact.entity.AreaOfLaw createNewAreaOfLawEntityFromModel(final AreaOfLaw areaOfLaw) {
+        final uk.gov.hmcts.dts.fact.entity.AreaOfLaw entity = new uk.gov.hmcts.dts.fact.entity.AreaOfLaw();
+        entity.setName(areaOfLaw.getName());
```

Comment:
Just wondering why are we setting the name within this method and setting all other fields in a separate method by passing in the newly created entity class. Again nothing with this just look a little bit strange like that. Alternatively can also create a constructor within the id and set them in one go using the constructor. I dont mind only really minor

Reply:
The name isn't editable in the admin portal, which is why it's like this. Something Ayisha picked up in testing is that if you try to change the name using the update endpoint, it returns a 200 and all fields updated are changed except the name. We perhaps could just check if the name is changed in the update request and return a bad request response if this is the case.